### PR TITLE
Prometheus: Add Victoria metrics to Prometheus type dropdown in datasource config

### LINKED ISSUE #66899: Prometheus: Add Victoria Metrics to prometheus type dropdown
**What is the chore?**:
Add victoria metrics to the dropdown options in prometheus data source config UI.

**Is there anything else we need to know?**:
Need to research if any VM versions don't support label/${name}/value endpoints with match[] parameters, and gate as needed.

### PR DESCRIPTION
**Pull request by gtk-grafana**
**What is this feature?**
Adding VictoriaMetrics to the prometheus "Type" dropdown in the [Prometheus data source config](https://grafana.com/docs/grafana/latest/datasources/prometheus/).

**Why do we need this feature?**
VM added support for match[] parameter in [v1.24.0](https://github.com/VictoriaMetrics/VictoriaMetrics/releases/tag/v1.24.0), right now VM users will be stuck using old /series endpoint which is more resource intensive, unless they select another datasource value that supports this API, which is a terribly confusing UX.

**Who is this feature for?**
Prometheus datasource users in Grafana that are running VictoriaMetrics.

**Which issue(s) does this PR fix?**:
Fixes https://github.com/grafana/grafana/issues/66899

**Special notes for your reviewer:**
The VM buildinfo api doesn't appear to give back any versioning info, at least not locally. For now the user will need to manually specify their VM version in the UI instead of being able to pull it from the API.

Please check that:
- [x] It works as expected from a user's perspective.
- [x] If this is a pre-GA feature, it is behind a feature toggle.
- [x] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Comment by gtk-grafana** [comment]
@bohandley heads up: looks like I'm making some changes that will be overwritten by your datasource config changes.

**Comment by bohandley** [comment]
> @bohandley heads up: looks like I'm making some changes that will be overwritten by your datasource config changes.

I'll keep an eye out for this on https://github.com/grafana/grafana/pull/66198

File: public/app/plugins/datasource/prometheus/datasource.tsx [hunk 1]
```diff
@@ -992,20 +994,23 @@ export class PrometheusDatasource
       [PromApplication.Mimir]: '/public/app/plugins/datasource/prometheus/img/mimir_logo.svg',
       [PromApplication.Prometheus]: '/public/app/plugins/datasource/prometheus/img/prometheus_logo.svg',
       [PromApplication.Thanos]: '/public/app/plugins/datasource/prometheus/img/thanos_logo.svg',
+      [PromApplication.VictoriaMetrics]: '/public/app/plugins/datasource/prometheus/img/vm_logo.svg',
```
Comment (gtk-grafana) [inline public/app/plugins/datasource/prometheus/datasource.tsx:997]: I'm adding this here, but the functionality that is rendering these images is being removed in https://github.com/grafana/grafana/pull/66198

Comment (bohandley) [reply public/app/plugins/datasource/prometheus/datasource.tsx:997]: Does this require a frontend testDataSource function? Was the rendering of the images dependent on that?

**Review by bohandley — APPROVED** [review]
It's looking good to me! 🚀✌🏻📈

File: public/app/plugins/datasource/prometheus/datasource.tsx (hunk 1 above)
Comment (gtk-grafana) [reply public/app/plugins/datasource/prometheus/datasource.tsx:997]: I'm not sure if I understand the question?

**Pull request merged**